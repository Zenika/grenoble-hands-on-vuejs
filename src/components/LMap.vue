<template>
  <div id="mapId"/>
</template>
<script>
import L from 'leaflet';

export default {
  name: "LMap",
  props: ['zoom', 'lat', 'long'],
  data() {
    return {
      map: null,
      zoomLevel: 13
    }
  },
  mounted() {
    this.map = L.map('mapId').setView([this.lat, this.long], this.zoomLevel)
    L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
      attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors',
      maxZoom: 18,
      tileSize: 512,
      zoomOffset: -1,
    }).addTo(this.map);
    L.marker([this.lat, this.long]).addTo(this.map)
  }
}
</script>
<style scoped>
#mapId {
  height: 350px;
  width: 100%;
}
</style>
