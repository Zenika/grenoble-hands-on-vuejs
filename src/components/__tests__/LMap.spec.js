import { shallowMount } from "@vue/test-utils";
import LMap from "@/components/LMap.vue";

jest.mock("leaflet", () => ({
  map: jest.fn(() => ({
    setView: jest.fn()
  })),
  tileLayer: jest.fn(() => ({
    addTo: jest.fn()
  })),
  marker: jest.fn(() => ({
    addTo: jest.fn()
  }))
}));
import L from 'leaflet';

describe("LMap.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(LMap, {
      propsData: {
        lat: 0,
        long: 0
      }
    });
  });
  it("should init map", () => {
    expect(wrapper.vm.map).not.toBeNull();
  });
  it("should zoom to level 13", () => {
    expect(wrapper.vm.zoomLevel).toBe(13);
  });
  test("if the template contains only a div", () => {
    expect(wrapper.findAll('div').length).toBe(1)
  });
  test("if the div has id mapId", () => {
    expect(wrapper.find('div').attributes('id')).toBe('mapId')
  });
  describe.skip('map leaflet', () => {
    beforeEach(() => {
      jest.clearAllMocks();
    });
    it('should register map on "mapId"', () => {
      expect(L.map).toHaveBeenCalledTimes(1);
      expect(L.map).toHaveBeenCalledWith('mapId');
    });
    it('should set zoom and lat/long', () => {
      // TODO
    });

    it('should set tileLayer', () => {
      expect(L.tileLayer).toHaveBeenCalledTimes(1);
      expect(L.tileLayer.mock.calls).toMatchSnapshot();
    });
    it('should add tileLayer to the map', () => {
      // TODO
    });

    it('should set marker', () => {
      expect(L.marker).toHaveBeenCalledTimes(1);
      expect(L.marker).toHaveBeenCalledWith([0, 0]);
    });
  });
});
