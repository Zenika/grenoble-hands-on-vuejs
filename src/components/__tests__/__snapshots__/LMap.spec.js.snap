// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LMap.vue map leaflet should set tileLayer 1`] = `
Array [
  Array [
    "http://{s}.tile.osm.org/{z}/{x}/{y}.png",
    Object {
      "attribution": "Map data &copy; <a href=\\"https://www.openstreetmap.org/\\">OpenStreetMap</a> contributors",
      "maxZoom": 18,
      "tileSize": 512,
      "zoomOffset": -1,
    },
  ],
]
`;
